### ISSUE TITLE
Cannot pull microsoft/aspnetcore - Failed to register layer

### ISSUE THREAD

**Issue by oskiFP** [Microsoft/aspnet-docker#71]
Pulling microsoft/aspnetcore image fails.

Execution of  `docker pull microsoft/aspnetcore` command results in:

```
Using default tag: latest
latest: Pulling from microsoft/aspnetcore
bce2fbc256ea: Already exists
53a0ccfb7e6f: Extracting [==================================================>]    148MB/148MB
05db938a42bb: Download complete
19a9c8f9472b: Download complete
f314af833261: Download complete
78e4cc48f3b5: Download complete
85a07057fe9a: Download complete
9be7ffeb20c4: Download complete
c2e6633a2cc5: Download complete
312abcf6e5bb: Download complete
4e3ebdf680a9: Download complete
failed to register layer: re-exec error: exit status 1: output: remove \\?\C:\ProgramData\Docker\windowsfilter\5f1cdf312dfc041dcf5ad4e548967b14c4995c9c8d4ddc2277baf9411c78562d\UtilityVM\Files\Windows\servicing\Packages\Microsoft-OneCore-Containers-Package~31bf3856ad364e35~amd64~~10.0.14393.0.cat: The process cannot access the file because it is being used by another process.
```

**System specification:**
CPU: AMD Athlon FX-4100
RAM: 12GB,
MOTHERBOARD: Gigabyte GA-970A-D3
OS: Windows 10 Pro

**Comment by lwsrbrts** [comment]
I have had this but not with this image. The issue seems to stem from the version being pulled when not using specific enough tags and the compatibility with the image that Docker decides to actually pull. Obviously this is all dependent on when you actually did the pull and your version of docker at the time, which you have provided above so I reckon this is explainable:

TL;DR - On the assumption you are still running that same version of Windows 10 and the same version of docker stable (why would you be?), the solution for you is likely to be to do `docker pull microsoft/aspnetcore:4.7.2-windowsservercore-ltsc`

So, why did this happen?

You're using Windows 10 1703, but when you did the docker pull, Docker (at version 17.09!) assumed you asked for a build of Windows Server 1709 (not to be confused with docker's version) because (that version of docker) will pull the latest OS version unless you specify which container OS version you want. Build 1709 of Windows Server (in the image) isn't backwards compatible with Build 1703 of Windows 10 (your OS). If you had updated docker to 17.10 (which was only released as an edge version), it would have pulled the ltsc image and saved you the hassle. See this issue and resolution for info:

Add support for Windows version filtering on pull - https://github.com/moby/moby/pull/35090

You're just a little unfortunate that you did a pull around this time and didn't have the latest (edge) version of docker that could detect compatibility and pull the right image. This note on the readme page is important in your context.

> Note: Multi-arch tags, such as 4.7.1, have been updated to use windowsservercore-1709 images if your host is Windows Server 1709 or higher or Windows 10 Fall Creators Update (Version 1709) or higher. You need Docker 17.10 or later to take advantage of these updated tags.

Hopefully that makes sense.
